google-cloud-spanner/src/test/java/com/google/cloud/spanner/DatabaseClientImplTest.java

```diff
@@ -3662,6 +3677,34 @@ public void testBatchCreateSessionsPermissionDenied() {
       // Actually trying to get any results will cause an exception.
       SpannerException e = assertThrows(SpannerException.class, rs::next);
       assertEquals(ErrorCode.PERMISSION_DENIED, e.getErrorCode());
+    } finally {
+      mockSpanner.setBatchCreateSessionsExecutionTime(SimulatedExecutionTime.none());
+    }
+  }
+
+  @Test
+  public void testCreateSessionsPermissionDenied() {
```

Comment:
I don't think you need separate tests for these conditions. `BatchCreateSessions` and `CreateSession` are both exclusively used by respectively non-multiplexed-sessions and multiplexed-sessions. That means that you can just register the same error for both, and depending on whether multiplexed sessions are enabled or not, one of the RPCs will be called, and the behavior should be the same in both cases.

Reply:
Right, modified to re-use the same test.